### PR TITLE
api(popup): introduce BrowserContext.exposeFunction

### PR DESCRIPTION
**Pull request by dgozman**

### CODE REVIEW

File: docs/api.md [hunk 1]
```diff
@@ -354,6 +355,73 @@ will be closed.
 If no URLs are specified, this method returns all cookies.
 If URLs are specified, only cookies that affect those URLs are returned.
 
+#### browserContext.exposeFunction(name, playwrightFunction)
+- `name` <[string]> Name of the function on the window object.
+- `playwrightFunction` <[function]> Callback function which will be called in Playwright's context.
+- returns: <[Promise]>
+
+The method adds a function called `name` on the `window` object of every page in the context.
```
Comment (pavelfeldman) [inline docs/api.md]: mention iframes?

File: docs/api.md [hunk 2]
```diff
@@ -354,6 +355,73 @@ will be closed.
...
+
+(async () => {
+  const browser = await firefox.launch();
+  const context = await browser.newContext();
+  await context.exposeFunction('md5', text =>
+    crypto.createHash('md5').update(text).digest('hex')
+  );
+  const page = await context.newPage();
+  page.on('console', msg => console.log(msg.text()));
+  await page.evaluate(async () => {
```
Comment (pavelfeldman) [inline docs/api.md]: It would be nice if md5 was used in a page context snippet.

File: src/page.ts [hunk 3]
```diff
@@ -598,3 +556,67 @@ export class Worker extends platform.EventEmitter {
     return (await this._executionContextPromise).evaluateHandle(pageFunction, ...args as any);
   }
 }
+
+export class PageBindng {
```
Comment (pavelfeldman) [inline src/page.ts]: Binding

File: test/evaluation.spec.js [hunk 4]
```diff
@@ -274,7 +274,15 @@ module.exports.describe = function({testRunner, expect, FFOX, CHROMIUM, WEBKIT})
       await page.evaluate(() => { window.JSON.stringify = null; window.JSON = null; });
       const result = await page.evaluate(() => ({abc: 123}));
       expect(result).toEqual({abc: 123});
-    })
+    });
+    it.skip(WEBKIT)('should await promise from popup', async function({page, server}) {
```
Comment (pavelfeldman) [inline test/evaluation.spec.js]: it.fail

File: test/page.spec.js [hunk 5]
```diff
@@ -385,6 +385,31 @@ module.exports.describe = function({testRunner, expect, headless, playwright, FF
       });
       expect(result).toBe(36);
     });
+    it.skip(CHROMIUM || FFOX)('should work with browser context', async({browser, server}) => {
```
Comment (pavelfeldman) [inline test/page.spec.js]: move this to browsercontext.spec.js

**Review by pavelfeldman — APPROVED** [review]

File: docs/api.md (hunk 1 above)
Comment (dgozman) [reply docs/api.md]: Done.

File: docs/api.md (hunk 2 above)
Comment (dgozman) [reply docs/api.md]: Done.

File: src/page.ts (hunk 3 above)
Comment (dgozman) [reply src/page.ts]: Done.

File: test/evaluation.spec.js (hunk 4 above)
Comment (dgozman) [reply test/evaluation.spec.js]: Done.

File: test/page.spec.js (hunk 5 above)
Comment (dgozman) [reply test/page.spec.js]: Done.

**Review by dgozman — COMMENTED** [review]
All done.

**Pull request merged**